go/enclave/enclave.go

```diff
@@ -1230,19 +1244,30 @@ func (e *enclaveImpl) produceBlockSubmissionResponse(l2Head *common.L2BatchHash,
 func (e *enclaveImpl) subscriptionLogs(upToBatchNr *big.Int) (map[gethrpc.ID][]byte, error) {
 	result := map[gethrpc.ID][]*types.Log{}
 
-	// Go through each subscription
+	// Go through each subscription and collect the logs
 	err := e.subscriptionManager.ForEachSubscription(func(id gethrpc.ID, subscription *common.LogSubscription, previousHead *big.Int) error {
 		// 1. fetch the logs since the last request
-		fromBlock := previousHead
-		// when the subscription is initialised, default from the latest batch
-		if previousHead == nil || previousHead.Int64() < 0 {
+		var from *big.Int
+		to := big.NewInt(upToBatchNr.Int64() + 1)
+
+		if previousHead == nil || previousHead.Int64() <= 0 {
+			// when the subscription is initialised, default from the latest batch
 			batch, err := e.storage.FetchHeadBatch()
 			if err != nil {
 				return err
 			}
-			fromBlock = batch.Number()
+			from = batch.Number()
+		} else {
+			from = big.NewInt(previousHead.Int64() + 1)
 		}
-		logs, err := e.storage.FilterLogs(subscription.Account, fromBlock, upToBatchNr, nil, subscription.Filter.Addresses, subscription.Filter.Topics)
+
+		if from.Cmp(to) >= 0 {
+			e.logger.Warn(fmt.Sprintf("Skipping subscription step id=%s: [%d, %d]", id, from, to))
+			return nil
+		}
+
+		e.logger.Info(fmt.Sprintf("Subscription id=%s: [%d, %d]", id, from, to))
```

Comment:
If you're leaving this log line in then maybe worth putting it after the FilterLogs call just below and including number of matching logs found? (With a warn or error log in the err check so we still see them). Would give us a bit more visibility for debugging.

Reply:
good idea